### ISSUE TITLE
Not working on IntelliJ 2024.3

### ISSUE THREAD

**Issue by wilkerlucio** [eeemt/AceJump-Lite#12]
Hi, I noticed that since I upgraded IntelliJ to 2024.3, the AceJump light has stopped working.

**Comment by schoettker** [comment]
Hey, same here and I'm not sure if this plugin is still maintained.
I have created a [fork](https://github.com/schoettker/AceJump-Reloaded) and updated the plugin to work with the new IntelliJ plugin ecosystem. 
It is published here: https://plugins.jetbrains.com/plugin/25997-acejump-reloaded

I'm not super familiar with Intellij Plugin development so I was just trying to get this working again, and it does work for me now as it used to but I was also not using every feature of the plugin - so if there's something not working for you, let me know and I can have a look. 

All credits for the source code and ground work to @EeeMt 🙏

**Comment by alexandr1369** [comment]
@schoettker Cool, thanks for a try! Looking forward to @EeeMt 's reply...

**Comment by arialdomartini** [comment]
@schoettker it works perfectly. Thank you!
Which issue was that? Would it be possible to submit a PR to the original plugin, or had you to modify the structure so much that this is in fact a separate plugin?

**Comment by schoettker** [comment]
It was a couple of things but I'd still be happy to submit this as a PR if there's an indication of the original plugin getting maintained going forward!
The main changes were:
- porting the plugin to be compatible with the [IntelliJ Platform Gradle Plugin (2.x)](https://plugins.jetbrains.com/docs/intellij/tools-intellij-platform-gradle-plugin.html)
  - this made building and developing the plugin easier, as there were not a lot of instructions how to develop the plugin locally
- fixing usages of illegal APIs that have been made illegal to use in more recent IntelliJ releases

The source code of the original plugin didn't change a lot and is actually itself mostly extracted from the [emacsIDEAs](https://github.com/whunmr/emacsIDEAs/tree/master) plugin.

One thing which is not supported/working yet in my fork is the custom configuration (which i personally haven't used even on the original plugin), but i will be looking into getting that to work. There's also another API usage marked to be deprecated by IntelliJ in upcoming releases so I will look into getting ahead of that 😊

**Comment by arialdomartini** [comment]
@schoettker oh no! Just noticed that because of the custom configuration has been removed, now the "Jump to behind" option is gone. To me, the standard behavior is very inconvenient.

I mean, I want to jump to the word "foo", I select "f" expecting to find the cursor at "_foo". Instead, it moves one char forward "f_oo".

Any chance to change this?

**Comment by schoettker** [comment]
@arialdomartini i agree! the default should probably be to jump to the very first character - feels more natural. I will set this as the default in the next release together with enabling the configuration options, so that people who rely on the current default behavior can still use that if they want 😊 I'm currently traveling for Christmas but i should probably get this done this coming weekend!

**Comment by arialdomartini** [comment]
Amazing! We could also move the conversation to your fork. Enjoy your Xmas!

**Comment by alexandr1369** [comment]
It doesn't work for me on MacOS... What's shortcut for this self-fixed version?

**Comment by schoettker** [comment]
@alexandr1369 did you bind an `AceJumpXY` action to any keys? There are no default keybindings. To check if it's generally working you can open the Actions Palette in IntelliJ and search for "AceJump" then you should be able to select e.g. `AceJumpWord`.
Your most used actions you can of course bind to some keys of your liking. 
<img width="816" alt="image" src="https://github.com/user-attachments/assets/097f673f-e4e0-4182-bdfa-b8fa4a104a06" />

**Comment by schoettker** [comment]
@arialdomartini, the update to jump before/onto the first character is now out for verification by Jetbrains (which can take up to 2 business days) and should be available soon!
Changing the color settings is somehow not persisted across IDE restarts, but that seems to have also been the case with the original plugin actually. The "jump behaviour" is correctly persisted across restarts.
<img width="797" alt="image" src="https://github.com/user-attachments/assets/8e8daa61-4941-4b65-83de-c790ef097ad3" />

If you encounter any issues with this you can also let me know here 😊  https://github.com/schoettker/AceJump-Reloaded/issues/1

**Comment by arialdomartini** [comment]
@schoettker you are my hero!

**Comment by alexandr1369** [comment]
@schoettker Hi! Is it still under review? Im eager to try it finally! 
But one question - will it work for Rider? 
Maybe it's the main problem it doesn't work for me as it should?

**Comment by schoettker** [comment]
@alexandr1369 the new version to support the configuration settings is still under review, i think they only do it on Mon-Fri 😆 
But i would expect it to work on Rider yes! Did you try launching the action `AceJumpWord`  manually from the action palette?

**Comment by alexandr1369** [comment]
@schoettker Ahhh... Thanks a lot!!! Lifesaver )
Perfectly working with Rider 2024.3+ (only letter selecting feature needed).

**Comment by schoettker** [comment]
You're welcome! The update for the settings configuration is now available as well! 
> only letter selecting feature needed

Is `AceJumpCharAction` maybe the behaviour you're looking for?
If not let me know (feel free to create another issue) and I can have a look 😊
